scripts/functional-tests.sh

```diff
@@ -6,7 +6,24 @@ set -ex
 # build a prompt string that includes the time, source file, line number, and function name
 export PS4='+$(date +"%Y-%m-%d %T") ${BASH_VERSION}:${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
 
-pip install .
+pip install -e .
+
+export TEST_DIR='__functional-test-data'
+export XDG_DATA_HOME="${TEST_DIR}/.local/share"
+export XDG_CONFIG_HOME="${TEST_DIR}/.config"
+export PACKAGE_NAME='instructlab'  # name we use of the top-level XDG directories for CLI data
+export ILAB_DATA_DIR="${XDG_DATA_HOME}/${PACKAGE_NAME}"
```

Comment:
This is incompatible with macOS. XDG is a Linux standard. Apple has a different layout for user data.

Reply:
@tiran I've updated everything to rely on `platformdirs` instead, so we aren't explicitly referencing XDG anymore.